src/controllers/dlgprefcontroller.cpp

```diff
@@ -311,15 +311,21 @@ void DlgPrefController::enumeratePresets(const QString& selectedPresetPath) {
 
     // qDebug() << "Enumerating presets for controller" << m_pController->getName();
 
+    // Check the text color of the palette for whether to use dark or light icons
+    QString iconsPath = ":/images/preferences/dark/";
+    if (!Color::isDimColor(palette().text().color())) {
+        iconsPath = ":/images/preferences/light/";
+    }
```

Comment:
According to the docs we could also use `QDir` to access the resource system (didn't test it though). If that works we can get rid of the string concatenation:

```suggestion
    QString iconsPath = QDir(":/images/preferences/dark/") ;
    if (!Color::isDimColor(palette().text().color())) {
        iconsPath.setPath(":/images/preferences/light/");
    }
```

Then you can build paths like this which looks nicer to me:
```cpp
const auto noPresetIcon = QIcon(iconsPath.filePath("ic_none.svg"));
```

Reply:
I'd do it like this:
``` c++
    QString iconsPath = ":/images/preferences/dark/";
    if (!Color::isDimColor(palette().text().color())) {
        iconsPath = ":/images/preferences/light/";
    }
    m_iconsPath = iconsPath;
```
and
``` c++
    treeItem->setIcon(0, QIcon(m_iconsPath.filePath(iconFile)));
```
?
Builds fine and works.
I suspect QDir::filePath() does the same string concatenation interanlly btw.